### ISSUE TITLE
where do i ge tthe ipa

### ISSUE THREAD

**Issue by InvisibleRasta** [arichornlover/uYouEnhanced#523]
### Have you read the FAQ?

- [X] Yes, I did read the FAQ

### Is there an existing issue/question for this?

- [X] I have searched the existing issues

### Do you think this is a bug?

✅ Yes, I believe this is a bug. I will open a new issue with the bug template

### My question

Been searching the youtube.ipa file and I cant find it anywhere.. all versions are outdated on google.

### Additional context

no

**Comment by KingGamingYT** [comment]
You have to dump and decrypt your own.

**Comment by Dynvsty** [comment]
You have to build it yourself

**Comment by TheNightRider12** [comment]
https://github.com/arichornlover/uYouEnhanced/wiki/Building
When building the IPA using Github Actions, make sure you untick "Upload Artifact" so it doesn't upload the IPA publicly, but you can still view it yourself as a draft. Delete the draft release when you are done.

**Comment by InvisibleRasta** [comment]
yeah well there is no guide on how to do the dump and decrypt. so i guess ill have to find another solution. thank you

**Comment by KingGamingYT** [comment]
> yeah well there is no guide on how to do the dump and decrypt. so i guess ill have to find another solution. thank you

Take a peak at the r/Jailbreak subreddit, they should have resources that'll help you get started.

**Comment by EASENCH** [comment]
> Decrypted iOS App IPA

Hi! I am new to github and I can´t find the "build and release" action. Where can I find it? Should I have to edit it or just paste it in " set up a workflow yourself"?

I still don´t get why the need for a decrypted IPA. Do we have to find a new decrypted IPA every week or just using the first one for several weeks and just "build and release" it? 

I also have a problem with Git LFS (because it is needed to upload a file that is more than 100Mb which is this case). Do I have to put it on my /git folder on Windows and then upload it and that is it?
Thanks in advance.

**Comment by EASENCH** [comment]
In decypt.day it says that the .ipa file for Youtube is from 2 months ago. Is this the one we should use or it has to be one from the last week?

**Comment by TheNightRider12** [comment]
> > Decrypted iOS App IPA
> 
> Hi! I am new to github and I can´t find the "build and release" action. Where can I find it? Should I have to edit it or just paste it in " set up a workflow yourself"?
> 
> I still don´t get why the need for a decrypted IPA. Do we have to find a new decrypted IPA every week or just using the first one for several weeks and just "build and release" it?
> 
> I also have a problem with Git LFS (because it is needed to upload a file that is more than 100Mb which is this case). Do I have to put it on my /git folder on Windows and then upload it and that is it? Thanks in advance.

Note: You do not need to actually clone or download the repository. This can be done all from Github's Website. 

So you first need to fork this repository under your own Github account. Click the "Fork" button on the page of this repo.
Then you need to upload the decrypted IPA to somewhere, like dropbox. You can find the decrypted IPA on https://armconverter.com/decryptedappstore for example. You first need to make an iOSGods account and then you can download the IPA from there. Just connect the iOSGods Account to armconverter. It's pretty straightforward. Once you download the decrypted IPA, then upload it to Dropbox. From there, go back to Github. Go into the uYouEnhanced repo that you forked into your own Github profile. Then go to the Actions tab. Then go to the Workflows section. Then click "Build and Release uYouEnhanced". Then click "Run Workflow" on the right. Copy and paste the Dropbox Link for the YouTube IPA into the field "The direct URL to the decrypted YouTube ipa". And make sure you un-check the field "Upload Artifact " so that only "Create a draft release" is selected. Then click "Run Workflow"

"I still don´t get why the need for a decrypted IPA. Do we have to find a new decrypted IPA every week or just using the first one for several weeks and just "build and release" it?" - You need the decrypted IPA to be able to merge the tweak into the app. It's really hard to explain why in simple terms but you don't really need to know the why. Once Github builds the uYouEnhanced IPA just download that and install that to your phone. When you need to update the app, download the new decrypted IPA from whichever site you want when the YouTube update comes out and re-do the steps. However, with one step added. On your forked version of this repo, go into it and click "Sync Fork" and the Update. That is to pull the latest changes from this repo into your own forked repo. The button to do that will be under the green "Code" button. Then go into "Actions" and build the IPA again using the same steps as before. But now you will be using the new updated IPA link under "The direct URL to the decrypted YouTube ipa"

**Comment by EASENCH** [comment]
> > > Decrypted iOS App IPA

> 
> "I still don´t get why the need for a decrypted IPA. Do we have to find a new decrypted IPA every week or just using the first one for several weeks and just "build and release" it?" - You need the decrypted IPA to be able to merge the tweak into the app. It's really hard to explain why in simple terms but you don't really need to know the why. Once Github builds the uYouEnhanced IPA just download that and install that to your phone. When you need to update the app, download the new decrypted IPA from whichever site you want when the YouTube update comes out and re-do the steps. However, with one step added. On your forked version of this repo, go into it and click "Sync Fork" and the Update. That is to pull the latest changes from this repo into your own forked repo. The button to do that will be under the green "Code" button. Then go into "Actions" and build the IPA again using the same steps as before. But now you will be using the new updated IPA link under "The direct URL to the decrypted YouTube ipa"


Thank you again for your great reply. Finally I could manage to get it working. Just 2 more things that I am wondering.
When looking at the Workflows here:
https://github.com/arichornlover/uYouEnhanced/actions
There are like 10-12 different workflows in the last 2 days. Why is this? I mean do we have to check everyday this page to see if he has changed the workflow because if we don´t change it to the last version it would not work again?

Also what would happen when arichornlover stops updating his workflows? How can we build and release the same workflows without being dependent on him? Is there a website to learn what we have to change in the code whenever youtube updates his software? Because maybe in some months he does not continue with his project and we will have to find a solution for it.

**Comment by TheNightRider12** [comment]
> > > > Decrypted iOS App IPA
> 
> > "I still don´t get why the need for a decrypted IPA. Do we have to find a new decrypted IPA every week or just using the first one for several weeks and just "build and release" it?" - You need the decrypted IPA to be able to merge the tweak into the app. It's really hard to explain why in simple terms but you don't really need to know the why. Once Github builds the uYouEnhanced IPA just download that and install that to your phone. When you need to update the app, download the new decrypted IPA from whichever site you want when the YouTube update comes out and re-do the steps. However, with one step added. On your forked version of this repo, go into it and click "Sync Fork" and the Update. That is to pull the latest changes from this repo into your own forked repo. The button to do that will be under the green "Code" button. Then go into "Actions" and build the IPA again using the same steps as before. But now you will be using the new updated IPA link under "The direct URL to the decrypted YouTube ipa"
> 
> Thank you again for your great reply. Finally I could manage to get it working. Just 2 more things that I am wondering. When looking at the Workflows here: https://github.com/arichornlover/uYouEnhanced/actions There are like 10-12 different workflows in the last 2 days. Why is this? I mean do we have to check everyday this page to see if he has changed the workflow because if we don´t change it to the last version it would not work again?
> 
> Also what would happen when arichornlover stops updating his workflows? How can we build and release the same workflows without being dependent on him? Is there a website to learn what we have to change in the code whenever youtube updates his software? Because maybe in some months he does not continue with his project and we will have to find a solution for it.

Hi. So you do not need to worry about the workflows. He doesn't have to build workflows. You will be doing that yourself  under Actions in your own repository that you made by forking this one. You just need to worry about the changes made to this github repository. The changes are made here https://github.com/arichornlover/uYouEnhanced/commits/main/. Whenever you decide to make a new IPA, you go to your fork of the repository and sync the changes from this repo into your own fork. Which will pull the latest commits from arichornlover's repo into yours. Then you go through the process of building the IPA again. You can make a new IPA whenever you want. When a new update to YouTube comes out or whenever you desire. Just make sure you sync the changes into your repository.

**Comment by EASENCH** [comment]
>You just need to worry about the changes made to this github repository. The changes are made here https://github.com/arichornlover/uYouEnhanced/commits/main/. 
That is the issue. What would happen if arichornlover stops doing those changes at certain point in the future? How do we know what changes we would have to make (the ones that he is doing in his repository)?
 
> Edit: I personally sync the changes that I see made once every couple of days. Then I will build a new IPA when YouTube updates it's app in the app store. So I build a new IPA once a week roughly. Unless there is a fix to an issue that get's addressed, then I will make a new one then.
Ah. Ok. So the minor changes that he does in his repository are just minimal changes that he decides to do for some options inside the youtube application (like changing the options for video resolution or things like this).
And the great changes come when youtube launches a new update and in this case youtube will stop working on our Ipads and we would have to find the new decrypted .ipa.

**Comment by TheNightRider12** [comment]
> > You just need to worry about the changes made to this github repository. The changes are made here https://github.com/arichornlover/uYouEnhanced/commits/main/.
> > That is the issue. What would happen if arichornlover stops doing those changes at certain point in the future? How do we know what changes we would have to make (the ones that he is doing in his repository)?
> 
> > Edit: I personally sync the changes that I see made once every couple of days. Then I will build a new IPA when YouTube updates it's app in the app store. So I build a new IPA once a week roughly. Unless there is a fix to an issue that get's addressed, then I will make a new one then.
> > Ah. Ok. So the minor changes that he does in his repository are just minimal changes that he decides to do for some options inside the youtube application (like changing the options for video resolution or things like this).
> > And the great changes come when youtube launches a new update and in this case youtube will stop working on our Ipads and we would have to find the new decrypted .ipa.

The first part is correct.
The second part is correct in a way. You don't HAVE to update the IPA when YouTube comes out with a new update. I do not know if they make older versions stop working after a while or not. I know this is the case for Pokemon Go, I do not know if this is the case for YouTube. I update the IPA when YouTube updates the app just for the heck of it.

**Comment by EASENCH** [comment]
Great to know! Thank you.
